src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +441,38 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine }, StringSplitOptions.None);
+    foreach (var headerElement in headerElements)
+    {
+        // Trim header for empty spaces
+        var header = headerElement.Trim();
+
+        // A header string must have atleast one semi-colon

```

Comment:
```suggestion
        // A header string must have atleast one colon

```
Colon-> :
Semi-colon-> ;

Reply:
Updated